draft-pauly-add-deer.md

```diff
@@ -65,8 +84,256 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+
+DEER:
+: Discovery of Equivalent Encrypted Resolvers. Refers to the mechanisms defined
+in this document.
+
+Encrypted Resolver:
+: A DNS resolver using any encrypted DNS transport. This includes currently defined
+mechanisms such as DoH and DoT as well as future mechanisms.
+
+Equivalent Encrypted Resolver:
+: An Encrypted Resolver which is considered to provide answers equivalent to a given 
+resolver. This equivalency can be authenticated with PKI.
+
+Unencrypted Resolver:
+: A DNS resolver using TCP or UDP port 53.
+
+# Discovery Mechanism
+
+DNS resolvers can advertise one or more Equivalent Encrypted Resolvers that offer
+equivalent services over encrypted channels and are controlled by the same entity 
+as itself. To do this, a resolver returns one or more SVCB records for "dns://resolver.arpa"
+with "ipv4hint" and/or "ipv6hint" set to a valid IP address and at least one of the the encrypted
+resolver name keys set to a value used by the associated encrypted DNS transport. These
+values are discussed in {{encryption-types}}.
+
+The TLS certificate used with the resolver name MUST have the IP addresses for each of its DNS
+endpoints within the SubjectAlternativeName field to allow the client to support authenticated
+discovery.
+
+## Authenticated Discovery from Unencrypted Resolvers {#bootstrapping}
+
+When a DNS client is configured with an Unencrypted Resolver IP address, it SHOULD query
+the resolver for SVCB records for "dns://resolver.arpa" before making other queries.
+
+A DNS client MUST validate the Equivalent Encrypted Resolver and the Unencrypted Resolver that advertised it are
+related, the client MUST check the SubjectAlternativeName field of the Equivalent Encrypted Resolver's
+TLS certificate for the Unencrypted Resolver's IP address and the advertised IP address for the
+Equivalent Encrypted Resolver. If both are present, the discovered Equivalent Encrypted Resolver MUST
+be used whenever the Unencrypted Resolver would have been used. Otherwise, the client MUST NOT use the
+discovered resolver and SHOULD suppress queries for Equivalent Encrypted Resolvers against
+this resolver for the TTL of the negative or invalid response and continue to use the original resolver.
+
+### Why both IP addresses are required
+
+It is imperative that DNS clients require the IP addresses of both the Unencrypted Resolver
+and the Equivalent Encrypted Resolver to be present in the TLS certificate.
+
+If either IP address is not verified, an attacker can craft an SVCB record that will send the client
+to a valid Encrypted Resolver that has no relationship with the Unencrypted Resolver.
+
+### Encrypted Resolver SVCB Key Names {#encryption-types}
+
+This document defines SVCB keys for discovering DoH and DoT Equivalent Encrypted Resolvers.
+Future versions or other documents may define additional keys for discovery of resolvers
+using other encrypted transports.
+
+#### SVCB Records for Equivalent Encrypted Resolvers using DoH
```

Comment:
We may not want to nest so deeply, but can clean up later

Reply:
Agreed. I have no attachment to the current structure versus a flatter one. I just chose something to prevent writer's deadlock. The same is true of in-section security comments versus a security consideration section.